ewatercycle/forcing/_marrmot.py

```diff
@@ -1,27 +1,30 @@
 """Forcing related functionality for marrmot"""
 
-from dataclasses import dataclass
 from pathlib import Path
+from typing import Optional
 
 from esmvalcore.experimental import get_recipe
 
-from .datasets import DATASETS
-from .default import DefaultForcing
 from ..util import get_time
-
-GENERATE_DOCS = """Marrmot does not have model specific options."""
-LOAD_DOCS = """
-Fields:
-    forcing_file: Matlab file that contains forcings for Marrmot models. See format forcing file in `model implementation <https://github.com/wknoben/MARRMoT/blob/8f7e80979c2bef941c50f2fb19ce4998e7b273b0/BMI/lib/marrmotBMI_oct.m#L15-L19>`_.
-"""
+from ._default import DefaultForcing
+from .datasets import DATASETS
 
 
-@dataclass
 class MarrmotForcing(DefaultForcing):
     """Container for marrmot forcing data."""
-
-    # Model-specific attributes (preferably with default values):
-    forcing_file: str = 'marrmot.mat'
+    def __init__(
+        self,
+        start_time: str,
+        end_time: str,
+        directory: str,
+        shape: str,
+        forcing_file: Optional[str] = 'marrmot.mat',
```

Comment:
By not using dataclass we can make model specific options required.
```suggestion
        forcing_file: str,
    ):
```

Having defaults for load_foreign model_info felt to much as a surprise as the whole idea of load_foreign is to be specific.

Reply:
I removed it for marrmot. Do you also want to remove them for other models? Sometimes I think it does actually make sense to have defaults, and it saves typing. E.g. for wflow I like that the default variable names are the CMOR names.

Reply:
When the model has a single standard value then we can make that the default. Like for wflow var names.
Most filenames are not standardized as they include time ranges or can be in different formats.
So I think the default for filenames should be removed.

What do you think? Only keep non-file defaults

Reply:
I'm fine with removing all defaults for filenames, but maybe we should do it in a separate PR. We still get issues in the tests from making `forcing_info` required while `shape` is not required in `__init__` (this allows for loading external forcing for which no shapefile is available). I'll make a new issue.

Reply:
https://github.com/eWaterCycle/ewatercycle/issues/110